main.go

```diff
@@ -36,15 +36,43 @@ func connectionChecker(peer smtpd.Peer) error {
 	return smtpd.Error{Code: 421, Message: "Denied"}
 }
 
+func addrAllowed(addr string, allowedAddrs []string) bool {
+	if allowedAddrs == nil {
+		return true
+	}
+
+	domidx := strings.LastIndex(addr, "@")
+	if domidx == -1 {
+		return false
+	}
+	domain := strings.ToLower(addr[domidx+1:])
+
```

Comment:
```suggestion
  addr = strings.ToLower(addr)
```

Reply:
Applied (earlier in the function, so `domain` is lowercased, too).